src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -1307,6 +1307,47 @@ public void stopIndex(
       }
     }
 
+    @Override
+    public void reloadIndexState(
+        ReloadIndexStateRequest request, StreamObserver<DummyResponse> responseObserver) {
+      if (globalState.getConfiguration().getStateConfig().useLegacyStateManagement()) {
+        responseObserver.onError(
+            Status.UNAVAILABLE.withDescription("legacy state not supported").asRuntimeException());
+        return;
+      }
+      try {
+        if (globalState.getConfiguration().getIndexStartConfig().getMode().equals(Mode.REPLICA)) {
+          String indexName = request.getIndexName();
+          try {
+            globalState.getIndexStateManager(indexName).load();
```

Comment:
The index should be detectable by checking if it exists in `globalState.getIndexNames()`.

I'm not sure what the behavior should be here.
1. Just always resync all global/index state
2. Sync only index state if it exists, otherwise sync all. Should an empty index name also trigger sync all?

Reply:
What do you think if I remove the indexName param from the request and reload global state every time this endpoint is triggered? Do as long as this endpoint is called, reload global state, reload all index states and start all indexes if they should be started by not started yet.

Reply:
Yeah, that sounds good.